### ISSUE TITLE
The corresponding part of Equation 2 in the code

### ISSUE THREAD

**Issue by yjh576** [OatmealLiu/class-iNCD#3]
I'm sorry. I didn't find the corresponding part of equation 2 (of this paper paper) in the code. In addition, there are cases where equation 11 and the code do not correspond to positive and negative signs, which one is correct?

**Comment by OatmealLiu** [comment]
Hi! Thanks for the questions and your interest in our work.

Regarding Eq. (2):
We do not use Eq. (2) in our method. Eq. (2) is the definition `Logits-level Knowledge Distillation Loss`. We give it in `Sec.3.1 Preliminaries` to help the reader to understand our design thinking and motivation of our `Feature-level Knowledge Distillation Loss (aka, L_{KD}^{feat}`, which is described in Eq. (11) of the paper. As for the implementation of Eq. (11), please refer the code between line [#1191](https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py#L1191) and line [#1194](https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py#L1194) of script [incd_ablation_expt.py](https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py).

Regarding the sign in Eq. (11):
Please refer the sign in the code. PS: function [train_IL_center(model, old_model, train_loader, labeled_eval_loader,  unlabeled_eval_loader, all_eval_loader, class_mean, class_sig, class_cov, args)](https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py#L1093) in  script [incd_ablation_expt.py](https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py) is the implementation of the method FRoST we present in the paper.

Thanks!

Best regards,
Miu

**Comment by yjh576** [comment]
Sorry, maybe I didn't express myself clearly. In Eq. (11), the sign of the loss function is a negative sign, however, in line https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py#L1194, the code is a positive sign. Compared to Eq. (11) in the article, their signs are not consistent.

**Comment by OatmealLiu** [comment]
Hi! The sign in Eq. (11) should be positive. The code is correct.

The goal is to regularize the backbone feature extractor by minimizing the p2-norm of the feature extracted by the old backbone (frozen) and the one extracted by the current backbone for the same sample (unlabeled).

Best regards,
Miu

> Sorry, maybe I didn't express myself clearly. In Eq. (11), the sign of the loss function is a negative sign, however, in line
> 
> https://github.com/OatmealLiu/class-iNCD/blob/b4011b0fdb6f36d6938fae83c992109c8e5fad23/incd_ablation_expt.py#L1194
> 
> , the code is a positive sign. Compared to Eq. (11) in the article, their signs are not consistent.

**Comment by yjh576** [comment]
Ok. I see. Thank you!
